src/Hamiltonians/geometry.jl

```diff
@@ -0,0 +1,241 @@
+"""
+    abstract type LatticeGeometry
+
+A `LatticeGeometry` controls which sites in an [`AbstractFockAddress`](@ref) are considered
+to be neighbours.
+
+Currently only supported by [`HubbardRealSpace`](@ref).
+
+# Available implementations
+
+* [`PeriodicBoundaries`](@ref)
+* [`HardwallBoundaries`](@ref)
+* [`LadderBoundaries`](@ref)
+
+# Interface to implement
+
+* `Base.size`: return the lattice size.
+* [`neighbour_site(::LatticeGeometry, ::Int, ::Int)`](@ref)
+* [`num_neighbours(::LatticeGeometry)`](@ref)
+"""
+abstract type LatticeGeometry end
+
+function Base.show(io::IO, geom::LatticeGeometry)
+    print(io, nameof(typeof(geom)), size(geom))
+end
+
+function BitStringAddresses.onr(add, geom::LatticeGeometry)
```

Comment:
```suggestion
"""
    onr(add::AbstractFockAddress, geom::LatticeGeometry)
Returns the occupation number representation of a Fock state address as an `SArray` 
with the shape of the lattice geometry `geom`. For composite addresses, a tuple
of `onr`s is returned.
"""
function BitStringAddresses.onr(add, geom::LatticeGeometry)
```
Alternatively, one could consolidate the information into a single docstring for `onr`.